### PR TITLE
Creating a "latest" webpage

### PR DESCRIPTION
**Pull request by ajkannan**
This allows other gcloud platform websites to have a stable link to include for gcloud-java

### CODE REVIEW

**Comment by jgeewax** [comment]
Can you maybe rebase (and squash) this a bit to show the important commits? "trying to push again" seems like it's might not be all that useful...

/cc @dhermes

**Comment by dhermes** [comment]
13 commits for such a small change seems crazy. It should be one commit.

Try this:

```bash
$ git remote add upstream git@github.com:GoogleCloudPlatform/gcloud-java.git
$ git fetch upstream
$ # Should be 3a7facca461572713958c59b27b225ce57039d6e
$ git branch latest-webpage-via-travis upstream/master
$ git checkout latest-webpage-via-travis
$ git merge --squash master
$ git commit  # Whatever you'd like to put in your commit message
$ git push origin master
```

Then reset your `master` branch to just mirror the `master` branch in upstream:

```
$ git checkout master
$ git reset --hard $(git log upstream/master -1 --pretty=%H)
$ git push --force origin master
```

----

You shouldn't make commits to `master` if you plan to maintain your fork as a fork.

File: src/site/resources/index.html [hunk 1]
```diff
@@ -62,7 +62,7 @@ <h1 class="logo"><img src="img/logo-full.svg" alt="Google Cloud Platform" /></h1
                         <pre>&lt;dependency&gt;
   &lt;groupId&gt;com.google.gcloud&lt;/groupId&gt;
   &lt;artifactId&gt;gcloud-java&lt;/artifactId&gt;
-  &lt;version&gt;0.0.6&lt;/version&gt;
+  &lt;version&gt;SITE_VERSION&lt;/version&gt;
```
Comment (dhermes) [inline src/site/resources/index.html]: Is this intended to be a template value? Is there special notation to be used, e.g. `{{SITE_VERSION}}`?

File: utilities/after_success.sh [hunk 2]
```diff
@@ -10,6 +10,29 @@ if [ "${TRAVIS_JDK_VERSION}" == "oraclejdk7" -a "${TRAVIS_BRANCH}" == "master" -
     mvn cobertura:cobertura coveralls:report
     mvn site-deploy -DskipTests=true --settings=target/travis/settings.xml
     mvn deploy -DskipTests=true -Dgpg.skip=true --settings target/travis/settings.xml
+
+    # Update "site/latest/index.html" to redirect to the newly generated website
+    git config --global user.name "Travis CI"
+    git config --global user.email ${CI_DEPLOY_USERNAME}@github.com
```
Comment (dhermes) [inline utilities/after_success.sh]: We (`gcloud-python`) just use

```
git config --global user.email "travis@travis-ci.org"
git config --global user.name "travis-ci"
```

File: utilities/after_success.sh [hunk 3]
```diff
@@ -10,6 +10,29 @@ if [ "${TRAVIS_JDK_VERSION}" == "oraclejdk7" -a "${TRAVIS_BRANCH}" == "master" -
...
+    mkdir tmp_gh-pages # Avoid clashing filenames when cloning repo
+    cd tmp_gh-pages
+    git clone https://github.com/GoogleCloudPlatform/gcloud-java.git
```
Comment (dhermes) [inline utilities/after_success.sh]: You could just

```
git clone https://github.com/GoogleCloudPlatform/gcloud-java.git tmp_gh-pages
cd tmp_gh-pages
```

also you can clone a branch only via

```
git clone --branch gh-pages --single-branch https://github.com/GoogleCloudPlatform/gcloud-java/
```

**Comment by ajkannan** [comment]
Sorry about that, I'm new here and wasn't really thinking about the number of commits.  I'll fix that.

File: utilities/after_success.sh [hunk 4]
```diff
@@ -10,6 +10,29 @@ if [ "${TRAVIS_JDK_VERSION}" == "oraclejdk7" -a "${TRAVIS_BRANCH}" == "master" -
...
+    cd gcloud-java
+    SITE_VERSION=$(mvn org.apache.maven.plugins:maven-help-plugin:2.1.1:evaluate -Dexpression=project.version | grep -Ev '(^\[|Download\w+:)')
```
Comment (dhermes) [inline utilities/after_success.sh]: I'm surprised `maven` doesn't have a way to just give you this (i.e. surprised you need `grep`s help).

File: utilities/after_success.sh [hunk 5]
```diff
@@ -10,6 +10,29 @@ if [ "${TRAVIS_JDK_VERSION}" == "oraclejdk7" -a "${TRAVIS_BRANCH}" == "master" -
...
+    echo "<html><head><meta http-equiv=\"refresh\" content=\"0; URL='http://GoogleCloudPlatform.github.io/gcloud-java/site/${SITE_VERSION}/index.html'\" /></head><body></body></html>" > site/latest/index.html
+    git add site/latest/index.html
+
+    # Update "Quickstart with Maven" block on landing page to reflect latest version
+    sed -i "s/SITE_VERSION/$SITE_VERSION/g" site/${SITE_VERSION}/index.html
+    git add site/${SITE_VERSION}/index.html
+
+    git commit -m "Updating to reflect latest website version"
+    git config --global push.default simple
+    git push --force --quiet "https://${CI_DEPLOY_USERNAME}:${CI_DEPLOY_PASSWORD}@github.com/GoogleCloudPlatform/gcloud-java.git" > /dev/null 2>&1
```
Comment (dhermes) [inline utilities/after_success.sh]: Do you really want to `--force` push here? If you have multiple simultaneous builds that will cause problems.

**Comment by dhermes** [comment]
General question: This looks like it replaces `SITE_VERSION` every time a new commit is pushed for a particular version. Isn't that a bad thing? In other words, isn't it the point of versions to provide a set of stable releases that don't change out from under a user?

**Comment by jgeewax** [comment]
I'd agree -- we probably want to not override new versions, and fail when we try to...?

**Comment by aozarov** [comment]
This issue tried to do 2 things:

1) when we generate a site page for a specific version (each version has its own site page) the generated page "Quick start with maven" section should include the correct version (associated with its pom.xml).

2) A latest page (that will be main "site" page in GH) will point to the latest release site.

Both option should apply *only* to release versions (typically our POM is in non-release, SNAPSHOT mode).

File: utilities/after_success.sh (hunk 3 above)
Comment (ajkannan) [inline utilities/after_success.sh]: I'll use the first one, I need the master branch to get the pom.xml for the version.

Comment (dhermes) [inline utilities/after_success.sh]: Travis already has `master` checked out for you, so you can just grab it from there.

File: utilities/after_success.sh (hunk 4 above)
Comment (ajkannan) [inline utilities/after_success.sh]: Yeah, I was too.  But it seems to be a common problem according to stack overflow

File: utilities/after_success.sh (hunk 5 above)
Comment (ajkannan) [inline utilities/after_success.sh]: removing it

**Comment by ajkannan** [comment]
I have an updated pull request here: https://github.com/GoogleCloudPlatform/gcloud-java/pull/139

**Pull request closed without merging**